### PR TITLE
Fix THREE `MTLLoader.getIndex` return type error

### PR DESCRIPTION
**Pull request by LanceMoe**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

As follows:

```typescript
    getIndex(materialName: string): number /* <- I fixed this */ {
      return this.nameLookup[materialName]; // `nameLookup` is type of `{[key: string]: number;}`, so the return value should be a `number`
    };
    getAsArray(): Material[] {
      let index = 0; // the `index` is a `number`

      for (const mn in this.materialsInfo) {
        if (!this.materialsInfo.hasOwnProperty(mn)) {
          continue;
        }
        this.materialsArray[index] = this.create(mn);
        this.nameLookup[mn] = index; // `index` in this array `nameLookup`, so `nameLookup` is a `{[key: string]: number;}`
        index++;
      }

      return this.materialsArray;
    };

```

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## three ([<kbd>unpkg</kbd>](https://unpkg.com/browse/three@latest/))
was missing the following properties:
1. wrap
1. isGeneratorFunction
1. mark
1. awrap
1. AsyncIterator

<details>
<summary>as well as these 35 other properties...</summary>
<p>async, keys, values, AxisHelper, BinaryTextureLoader, BoundingBoxHelper, CanvasRenderer, ClosedSplineCurve3, CustomToneMapping, DynamicBufferAttribute, EdgesHelper, Face3, Face4, FaceColors, JSONLoader, LensFlare, Math, MeshFaceMaterial, MultiMaterial, NoColors, Particle, ParticleBasicMaterial, ParticleSystem, ParticleSystemMaterial, PointCloud, PointCloudMaterial, SceneUtils, Spline, SplineCurve3, Vertex, VertexColors, WebGLRenderTargetCube, WebGLUtils, WireframeHelper, XHRLoader</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a0fadf39288e37c86fb6e9e5bfad28da4147ec67
</p>

**Comment by joshuaellis** [comment]
Hi @LanceMoe, thanks for the contribution. 

Two things. Firstly, the current repository for the types lie in [`three-ts-types`](https://github.com/three-types/three-ts-types), so you're welcome to move your PR there? If not no worries, I can do the manual move for you after we merge this PR. 

Secondly, it would be really helpful if you wouldn't mind adding a test for `MTLLoader`, we've been copying the examples from the `three` repo. Here's an example of the [`GTLFLoader` test](https://github.com/three-types/three-ts-types/pull/10/files#diff-765a9ed49738217818824be319e59fadf699ad34d1a95e2bf181f4970c17cbd2R1).

**Comment by LanceMoe** [comment]
> Hi @LanceMoe, thanks for the contribution.
> 
> Two things. Firstly, the current repository for the types lie in [`three-ts-types`](https://github.com/three-types/three-ts-types), so you're welcome to move your PR there? If not no worries, I can do the manual move for you after we merge this PR.
> 
> Secondly, it would be really helpful if you wouldn't mind adding a test for `MTLLoader`, we've been copying the examples from the `three` repo. Here's an example of the [`GTLFLoader` test](https://github.com/three-types/three-ts-types/pull/10/files#diff-765a9ed49738217818824be319e59fadf699ad34d1a95e2bf181f4970c17cbd2R1).

Thanks for your reply, if you think my patch is ok, could you merge it please? And feel free to move this PR, I don't mind, I just want to help solve this problem. 😊
Now I am a beginner of THREE, I just encountered some problems in use and tried to solve, so I may not be able to write unit tests very correctly. When I can fully understand the code, I am willing to write and contribute some test code. 😜

**Review by joshuaellis — APPROVED** [review]
LGTM. I will copy these changes to `three-ts-types`.

**Comment by orta** [comment]
OK, my read on this is that the changes have been moved into the three js types repo, and are OK to be merged into here because they are now consistent. I'll leave it to you two to self-merge if that's correct 👍🏻

**Review by orta — APPROVED** [review]

**Comment by joshuaellis** [comment]
Ready to merge

**Pull request merged**
